api/views.py

```diff
@@ -147,6 +169,282 @@ def get_queryset(self):
+        real phone number purchase. If you do, you need to pass one of the
+        [test phone numbers][test-numbers].
+
+        The `number` should be in [E.164][e164] format.
+
+        Every call or text to the relay number will be sent as a webhook to the
+        URL configured for your `TWILIO_SMS_APPLICATION_SID`.
+
+        [test-creds]: https://www.twilio.com/docs/iam/test-credentials
+        [test-numbers]: https://www.twilio.com/docs/iam/test-credentials#test-incoming-phone-numbers-parameters-PhoneNumber
+        [e164]: https://en.wikipedia.org/wiki/E.164
+        """
+        return super().create(request, *args, **kwargs)
+
+    @decorators.action(detail=False)
+    def suggestions(self, request):
+        """
+        Returns suggested relay numbers for the authenticated user.
+
+        Based on the user's real number, returns available relay numbers:
+          * `same_prefix_options`: Numbers that match as much of the user's real number as possible.
+          * `other_areas_options`: Numbers that exactly match the user's real number, in a different area code.
+          * `same_area_options`: Other numbers in the same area code as the user.
+        """
+        numbers = suggested_numbers(request.user)
+        return response.Response(numbers)
+
+    @decorators.action(detail=False)
+    def search(self, request):
+        """
+        Search for available numbers.
+
+        This endpoints uses the underlying [AvailablePhoneNumbers][apn] API.
+
+        Accepted query params:
+          * ?location=
+            * Will be passed to `AvailablePhoneNumbers` `in_locality` param
+          * ?area_code=
+            * Will be passed to `AvailablePhoneNumbers` `area_code` param
+
+        [apn]: https://www.twilio.com/docs/phone-numbers/api/availablephonenumberlocal-resource#read-multiple-availablephonenumberlocal-resources
+        """
+        location = request.query_params.get("location")
+        if location is not None:
+            numbers = location_numbers(location)
+            return response.Response(numbers)
+
+        area_code = request.query_params.get("area_code")
+        if area_code is not None:
+            numbers = area_code_numbers(area_code)
+            return response.Response(numbers)
+
+        return response.Response({}, 404)
+
+
+def _validate_number(request):
+    parsed_number = _parse_number(request.data["number"], request)
+    if not parsed_number:
+        country = None
+        if hasattr(request, "country"):
+            country = request.country
+        error_message = f"number must be in E.164 format, or in local national format of the country detected: {country}"
+        raise exceptions.ValidationError(error_message)
+
+    e164_number = f"+{parsed_number.country_code}{parsed_number.national_number}"
+    number_details = _get_number_details(e164_number)
+    if not number_details:
+        raise exceptions.ValidationError(f"Could not get number details for {e164_number}")
```

Comment:
Hmm ... I can't think of a way we can know for sure if `+1503` meaning "country code 1 (US) area code 503" and `+503` meaning "country code 503 (El Salvador)". I.e., "+503xxxyyyy" *IS* a valid e.164 number format.

Also, this `_get_number_details` call failure is more generic like "Twilio couldn't return number details" so it might not have anything to do with e.164. So I suggest just keeping the error message as-is.